contracts/meta-oracles/DriftlessHistoricalPriceFeed.sol

```diff
@@ -0,0 +1,277 @@
+     * otherwise linearize current oracle price with last logged price to attempt to reduce potential error.
+     *
+     * @returns                     Price to update LinkedList with                  
+     */
+    function determineUpdatePrice()
+        private
+        returns (uint256)
+    {
+        // Get current medianizer value
+        uint256 medianizerValue = uint256(medianizerInstance.read());
+
+        // Since the nextAvailableUpdate timestamp was already update to prevent reentrancy we must subtract
+        // the updateFrequency then add the updateTolerance to get the timestamp after which we linearize
+        // the prices.
+        uint256 updateToleranceTimestamp = nextAvailableUpdate.sub(updateFrequency).add(updateTolerance);
+
+        // If block timestamp is greater than updateToleranceTimestamp we linearize the current price to try to
+        // reduce error
+        if (block.timestamp < updateToleranceTimestamp) {
+            return medianizerValue;
+        } else {
+            return linearizeDelayedPriceUpdate(medianizerValue);
+        }
+    }
+
+    /*
+     * When price update is delayed past the updateTolerance in order to attempt to reduce potential error
+     * linearize the price between the current time and price and the last updated time and price. This is 
+     * done with the following series of equations, modified in this instance to deal handle unsigned integers:
+     *
+     * updateTimeFraction = (updateFrequency/(block.timestamp - previousUpdateTimestamp))
+     *
+     * linearizedPrice = previousLoggedPrice + updateTimeFraction * (currentPrice - previousLoggedPrice)
+     *
+     * Where updateTimeFraction represents the fraction of time passed between the last update and now, spent in
+     * the previous update window. It's worth noting that because we consider updates to occur on their update
+     * timestamp we can make the assumption that the amount of time spent in the previous update window is equal
+     * to the update frequency. 
+     *
+     * @param  _currentPrice        Current price returned by medianizer
+     * @returns                     Linearized price value                  
+     */
+    function linearizeDelayedPriceUpdate(
+        uint256 _currentPrice
+    )
+        private
+        returns(uint256)
+    {
+        // Calculate the previous updates timestamp, due to already updating the timestamp to prevent reentrancy
+        // we must subtract two updateFrequency time periods
+        uint256 previousUpdateTimestamp = nextAvailableUpdate.sub(updateFrequency.mul(2));
+        // Calculate how much time has passed from last update
+        uint256 timeFromLastUpdate = block.timestamp.sub(previousUpdateTimestamp);
+
+        // Get previous price and put into uint256 format
+        uint256[] memory previousLoggedPriceArray = readList(historicalPriceData, 1);
```

Comment:
Could you destructure in 1 step? 

[previousLoggedPrice] = readList

Reply:
I don't believe you can do that in solidity. I may be wrong but trying it out I couldn't find a way to make it work.